invenio_users_resources/services/schemas.py

```diff
@@ -52,6 +52,8 @@ class UserSchema(BaseRecordSchema, FieldPermissionsMixin):
         "active": "read_details",
         "confirmed": "read_details",
         "preferences": "read_details",
+        "blocked_at": "read_system_details",
```

Comment:
Question: They are not serialised out to normal users right? Only admins should see these details.

Reply:
Yes, these fields' visibility is computed based on permissions.
I created a new permission that allows only user managers and system.

I added [tests](https://github.com/inveniosoftware/invenio-users-resources/pull/91/files#diff-78bf3e2cf2f12f9e039080bc98f06b41e86ab016328af6fe201d843f7b5daf49R161) for this:

```python
  blocked = user_service.block(user_moderator.identity, user_res.id)
  assert blocked

  ur = user_service.read(user_res.identity, user_res.id)
  # User can't see when it was blocked
  assert not "blocked_at" in ur.data
  # But can see it's not active
  assert ur.data["active"] is False

  ur = user_service.read(user_moderator.identity, user_res.id)
  # Moderator can see the blocked_at time
  assert ur.data["blocked_at"] is not None
```

Reply:
Have you tried with a public user profile?

Reply:
> Have you tried with a public user profile?

What do you mean? The permissions on each field are tweakable by field. E.g. `blocked_at` is only available to whoever has the `read_system_details` permissions. In this case:

```python
can_read_system_details = [UserModeration(), SystemProcess()]
```